### PR TITLE
Fix rendering translations in JS

### LINKED ISSUE #5459: JS strings are not translated
As reported in a [support topic](https://wordpress.org/support/topic/cannot-be-translated-the-initials-settings-page/), the strings in JS are not being translated, or at least they aren't getting loaded properly.

When I activate `es-MX`, I see the admin screen as:

> image

This is in spite of this language being supposedly 100% translated:

> image

Either the strings aren't getting translated, or we're not loading the translations properly.

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

Fixes #5459 

Per @swissspidy in https://github.com/ampproject/amp-wp/issues/5459#issuecomment-703490232:

String extraction and actual loading of translations works fine, as can be tested by typing `wp.i18n.__( 'Reopen Wizard', 'amp')` in the console. The problem is that the settings page is bundling `@wordpress/i18n` and not actually using `wp.i18n`.

`@wordpress/i18n` needs to be added as an external in the webpack config, so that calls to `__()` in the code base are mapped to `window.wp.i18n.__()`. Then it will work.

In my quick testing, this change to the settings page config worked:

```diff
diff --git webpack.config.js webpack.config.js
index aab58f6ab..d4571c6ea 100644
--- webpack.config.js
+++ webpack.config.js
@@ -234,6 +234,7 @@ const setup = {
 					case '@wordpress/dom-ready':
 					case '@wordpress/html-entities':
 					case '@wordpress/url':
+					case '@wordpress/i18n':
 						return defaultRequestToHandle( handle );
 
 					default:
@@ -247,6 +248,7 @@ const setup = {
 					case '@wordpress/dom-ready':
 					case '@wordpress/html-entities':
 					case '@wordpress/url':
+					case '@wordpress/i18n':
 						return defaultRequestToExternal( external );
 
 					default:
@@ -293,6 +295,7 @@ const settingsPage = {
 				switch ( handle ) {
 					case 'lodash':
 					case '@wordpress/api-fetch':
+					case '@wordpress/i18n':
 						return defaultRequestToHandle( handle );
 
 					default:
@@ -303,6 +306,7 @@ const settingsPage = {
 				switch ( external ) {
 					case 'lodash':
 					case '@wordpress/api-fetch':
+					case '@wordpress/i18n':
 						return defaultRequestToExternal( external );
 
 					default:
```

This needs to be checked for all other places too.

## Checklist

- [ ] My pull request is addressing an open issue (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by westonruter** [comment]
Works on settings screen:

Settings screen

**Comment by westonruter** [comment]
Is not working in the onboarding wizard:

image

**Comment by johnwatkins0** [comment]
It seems like the onboarding wizard strings aren't being picked up at all when I look around at https://translate.wordpress.org/locale/de/default/wp-plugins/amp/ 

I'll look into why that might be.

**Comment by westonruter** [comment]
I would have thought that strings in common with the settings screen would at least be loaded:

image

**Comment by johnwatkins0** [comment]
The JS translations are associated with their source file. E.g., the one for the `amp-settings.js` file is easily found in the list of language files:

Screen Shot 2020-10-05 at 1 34 13 PM

When printing the onboarding wizard scripts, `load_script_textdomain` goes looking for the corresponding file associated with `assets/js/amp-onboarding-wizard.js`, but that file is not there. 

Translations within the `amp-onboarding-wizard.js` entry point are treated exactly the same as the other JS files in the plugin, so I'm wondering what about this file makes it not show up in the translation tools.

**Comment by pierlon** [comment]
What's interesting is that the translations from the onboarding wizard JS are included when generated by WP-CLI (command: `wp i18n make-pot . languages/test.pot`). I'm not sure if WP-CLI is also used by WP to generate and parse translations as well.

**Comment by johnwatkins0** [comment]
@pierlon Is that on `develop` or on this PR branch with the JS rebuilt?

**Comment by pierlon** [comment]
In both branches. Rebuilding the JS with the current changes in this PR shouldn't change how the translations are parsed, would it?

**Comment by johnwatkins0** [comment]
Agreed, I don't think that would matter. Just checking.

**Comment by schlessera** [comment]
WP-CLI has a lot of custom logic to extract JS strings, so it might indeed be a case of the .org infrastructure not yet being up-to-date and requiring us to manually provide the translation files.

@swissspidy built most of the WP-CLI i18n tooling, so he should know!

**Comment by swissspidy** [comment]
@pierlon you need to run the script in the build folder, not the source folder. Did you do that?

**Comment by pierlon** [comment]
@swissspidy  Oh no, I did not. That may be the reason why I got the onboarding wizard translations while on the `develop` branch.

---

I just ran the script from the build folder (from this branch) and it did manage to extract the strings from the onboarding wizard JS. And as should be expected, the strings were not extracted when I ran the script from the `develop` branch. So once the external `wp.i18n` is used the strings are correctly detected and extracted, as expected.

**Comment by johnwatkins0** [comment]
> So once the external `wp.i18n` is used the strings are correctly detected and extracted, as expected.

In that case, I think it's a safe bet that this PR will fix it.

**Comment by westonruter** [comment]
I _am_ seeing JS translation in the edit post screen:

image

Also in the AMP Customizer for a Reader theme:

image

**Comment by westonruter** [comment]
@swissspidy I did an `npm run build:prod` and then I went into `build/`. In the built `amp-validated-url-post-edit-screen.js`, I can see `__()` being built as `Object(a.__)("You have unsaved changes. Are you sure you want to leave?","amp")`. Is that as expected?

**Comment by pierlon** [comment]
@westonruter That's expected as far as I know. You can see in the WP CLI command that that's the syntax it's expecting: https://github.com/wp-cli/i18n-command/blob/b02ecdc9a57f9633740c254d19749118b7411f0f/src/JsFunctionsScanner.php#L195-L196.

**Comment by westonruter** [comment]
That's saying:

```
For example: wp.i18n.__( "translation" ) or u.__( "translation" ).
```

But here we have:

```
Object(a.__)("You have unsaved changes. Are you sure you want to leave?","amp")
```

Ah, but [below](https://github.com/wp-cli/i18n-command/blob/b02ecdc9a57f9633740c254d19749118b7411f0f/src/JsFunctionsScanner.php#L212-L213) it shows this as an example:

```
// If the callee is a call expression as created by Webpack resolve it.
// For example: Object(u.__)( "translation" ).
```

**Comment by westonruter** [comment]
I'll merge once you and @johnwatkins0 approve.

**Review by pierlon — APPROVED** [review]
Learnt a lot from this. Changes look good to me :+1:.

**Review by johnwatkins0 — APPROVED** [review]
This looks like it will fix the missing translations. 👍

**Pull request merged**
